internal/xsql/sqlValidator.go

```diff
@@ -31,5 +31,96 @@ func Validate(stmt *ast.SelectStatement) error {
 			return fmt.Errorf("Not allowed to call aggregate functions in GROUP BY clause.")
 		}
 	}
+
+	if err := validateSRFNestedForbidden("select", stmt.Fields); err != nil {
+		return err
+	}
+	if err := validateMultiSRFForbidden("select", stmt.Fields); err != nil {
+		return err
+	}
+	if err := validateSRFForbidden("where", stmt.Condition); err != nil {
```

Comment:
Can we just check not fields case. Pseudo code below:
```
if srf clause is not fields
then
   return validation error
end

Otherwise, we'll need to modify here each time we add new clause type.
```

Reply:
updated.